From 3aab4ad34b8a9b5b87f80a5a11170c4fd79e49dd Mon Sep 17 00:00:00 2001
From: Justin Stephenson <jstephen@redhat.com>
Date: Tue, 26 Sep 2023 16:24:16 -0400
Subject: [PATCH] Passkey: Increase conv message size for prompting

Size needs to handle the prompts for interactive, touch, pin prompt, and
kerberos pre-auth warning message which could all be displayed.
---
 src/sss_client/pam_sss.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/sss_client/pam_sss.c b/src/sss_client/pam_sss.c
index 2cc43c3ff7..a1c3536048 100644
--- a/src/sss_client/pam_sss.c
+++ b/src/sss_client/pam_sss.c
@@ -1849,8 +1849,8 @@ static int prompt_passkey(pam_handle_t *pamh, struct pam_items *pi,
 {
     int ret;
     const struct pam_conv *conv;
-    const struct pam_message *mesg[3] = { NULL, NULL, NULL };
-    struct pam_message m[3] = { {0}, {0}, {0} };
+    const struct pam_message *mesg[4] = { NULL, NULL, NULL, NULL };
+    struct pam_message m[4] = { {0}, {0}, {0}, {0} };
     struct pam_response *resp = NULL;
     bool kerberos_preauth;
     bool prompt_pin;


